src/garage/envs/dm_control/dm_control_env.py

```diff
@@ -8,33 +10,82 @@
 from garage.envs.dm_control.dm_control_viewer import DmControlViewer
 
 
-class DmControlEnv(gym.Env):
-    """
-    Binding for `dm_control <https://arxiv.org/pdf/1801.00690.pdf>`_
+def _flat_shape(observation):
+    """Returns the flattend shape of observation.
+
+    Args:
+        observation (np.ndarray): the observation
+
+    Returns:
+        np.ndarray: the flattened dimension of observation.
     """
+    return np.sum(int(np.prod(v.shape)) for k, v in observation.items())
+
+
+class DmControlEnv(gym.Env):
```

Comment:
why not just make it garage.Environment if you're going to touch so much code?

Reply:
I can't because changes of TimeStep are not merged yet. I touched the code for exercise and later reverted the changes. I have to fix the style to commit it.

I'd prefer to leave dm_control_env as it is, then rewrite it with the new API in a new PR.